imports/api/users.js

```diff
@@ -59,11 +61,24 @@ _.extend(User.prototype, {
     else return false
   },
   getImageUrl: function () {
-    return this.profile.profileImage ? this.profile.profileImage : '/images/avatar.png'
+    if (this.profile.profileThumbnail) {
+      const settings = Settings.findOne()
+      if ( settings && settings.storageType === 'AWS' && !this.profile.profileImage.endsWith('/image')) {
+        Meteor.call('users.updateProfileImage', this.profile.profileImage + '/image', settings.storageType)
```

Comment:
First, check if *not* this.profile.profileThumbnail, since the updated code for setting the profile will set the thumbnail  and profile to an actual URL.

Then, the condition to make an update is:

if(this.profile.profileImage.beginsWith('https://ql-images-1.s3-ca-central-1.amazonaws.com') && !this.profile.profileImage.endsWith('/image') )

Also make the update to the thumbnailURL here (create it if it does not exist). 

Don't check the settings, since those are irrelevant to pictures updated in the past.